### ISSUE TITLE
SemanticMasks to Image - Diffusion Model

### ISSUE THREAD

**Issue by FrsECM** [huggingface/diffusers#5321]
Hi Everyone !

**Is your feature request related to a problem? Please describe.**
I've read the documentation and i saw that there is theses task implemented :
Unconditional Image Generation
Text-Guided Image Generation
Text-Guided Image-to-Image Translation
Text-Guided Image-Inpainting
Text-Guided Depth-to-Image Translation

In order to improve semantic segmentation pipelines, i'd like to **generate synthetic data** from **semantic masks** in order to reduce the number of edge cases in our supervised datasets.
(move foregrounds on different background etc....) 
We wouldn't be in open vocabulary.

**Describe the solution you'd like**
For that, i'd like to condition the generation with a semantic mask, like in this paper :
https://arxiv.org/abs/2207.00050

**Describe alternatives you've considered**
I tried to use open vocabulary text to image generation but there is two main issues :
- The CLIP model is badly adapted to very specific issues that can face private companies / industrials then images are rarely releavant.
- We need to retrain VAE because sometime the input distribution of images is very far from open worlds one.

**Additional context**
New in Hugging face, i'd need some insights in order to get started.

**Do somebody already implemented such things within the framework ?**

Intuitively i'd say that :
I need to train a custom VAE in order to fit with my own images (very specific 2 channels images)
I need to modify the unet conditionning in order to take as input the semantic mask.

For VAE :
It there already a training pipeline for that ?

For UNet :
How to condition it with an image instead of a text embedding ?

For pipeline : 
Maybe implement a StableDiffusionSem2ImgSPipeline
Because StableDiffusionImg2ImgPipeline requires on a textencoder/tokenizer that are not necessary here.

Thanks for your insights !

**Comment by DN6** [comment]
@FrsECM Would you consider adding the pipeline to the community pipelines?

**Comment by DN6** [comment]
@FrsECM Of course. Please open a PR and we can review.

**Comment by FrsECM** [comment]
Hi @DN6  !
I've created a local branch. I didn't sync it for the moment.

I'm working on the training script and i have a misunderstanding of the paper (i'm discovering diffusion for this project).
I base what i do on that [code ](https://github.com/WeilunWang/semantic-diffusion-model/blob/main/guided_diffusion/script_util.py)and that [paper](https://arxiv.org/abs/2207.00050).
In the paper, they define the diffusion process like that :  
<img src='https://github.com/huggingface/diffusers/assets/26071804/59f6f795-12d3-4071-a341-b5e8727b5d40' height=200/>
 
And they say this : 
<img src='https://github.com/huggingface/diffusers/assets/26071804/d6bd14e8-5f73-4b79-a2e6-28020e1bd489' height=200/>

We know : 
$q(y_{t}|y_{t-1},y_{0})$ is known **but** $q(y_{t-1}|y_{t},y_{0})$ is unknown ?

**Then how to compute the KL in this case ?**
Should i ignore this ?

Thanks for your help !

**Comment by FrsECM** [comment]
@DN6, I created the pull request there :
https://github.com/huggingface/diffusers/pull/5431

Fill free to review it. I need support especially there : 

https://github.com/FrsECM/diffusers/blob/8bd3a3fe3c9b0679ec0fc6078befc024c636aad4/examples/semantic_image_synthesis/train_sis.py#L288-L303
